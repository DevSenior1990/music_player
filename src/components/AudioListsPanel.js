import React, { Fragment } from "react";
import cls from "classnames";
import NotContent from "react-icons/lib/md/library-music";

const AudioListsPanel = ({
  audioLists,
  visible,
  notContentText,
  onCancel,
  onDelete,
  onPlay,
  pause,
  playId,
  loading,
  playIcon,
  pauseIcon,
  closeIcon,
  deleteIcon,
  isMobile,
  panelTitle,
  panelToggleAnimate,
  glassBg,
  cover,
  remove
}) => (
  <div
    className={cls("audio-lists-panel", panelToggleAnimate)}
    key="audio-list-panel"
  >
    {glassBg ? (
      <div
        className="glass-bg-container"
        style={{ backgroundImage: `url(${cover})` }}
      />
    ) : (
      undefined
    )}
    <div className="audio-lists-panel-header">
      <h2 className="title">
        <span key="panel-title">{panelTitle} / </span>
        <span className="num" key="num">
          {audioLists.length}
        </span>
        <span
          key="close-btn"
          className="close-btn"
          title="close"
          {...{ [isMobile ? "onTouchStart" : "onClick"]: onCancel }}
        >
          {closeIcon}
        </span>
        {remove ? (
          <Fragment>
            <span className="line" key="line" />
            <span
              key="delete-btn"
              className="delete-btn"
              title="delete audio lists"
              {...{ [isMobile ? "onTouchStart" : "onClick"]: onDelete }}
            >
              {deleteIcon}
            </span>
          </Fragment>
        ) : (
          undefined
        )}
      </h2>
    </div>
    <div
      className={cls("audio-lists-panel-content", {
        "no-content": audioLists.length < 1
      })}
      key="audio-lists-panel-content"
    >
      {audioLists.length >= 1 ? (
        <ul>
          {audioLists.map((audio, i) => {
            const { name, singer } = audio;
            return (
              <li
                key={i}
                title={
                  pause
                    ? "Click to play"
                    : playId === i
                      ? "Click to pause"
                      : "Click to play"
                }
                className={cls(
                  "audio-item",
                  { playing: playId === i },
                  { pause }
                )}
                {...{
                  [isMobile ? "onTouchStart" : "onClick"]: () => onPlay(i)
                }}
              >
                <span className="group player-status" key="player-status">
                  <span className="player-icons" key={`player-icons-${i}`}>
                    {playId === i && loading
                      ? loading
                      : playId === i
                        ? pause
                          ? playIcon
                          : pauseIcon
                        : undefined}
                  </span>
                </span>
                <span className="group player-name" key="player-name">
                  {name}
                </span>
                <span className="group player-time" key="player-time">
                  {singer}
                </span>
                {remove ? (
                  <span className="group player-delete" key="player-delete">
                    {closeIcon}
                  </span>
                ) : (
                  undefined
                )}
              </li>
            );
          })}
        </ul>
      ) : (
        <Fragment>
          <span>
            <NotContent />
          </span>
          <span className="no-data" key="no-data">
            {notContentText}
          </span>
        </Fragment>
      )}
    </div>
  </div>
);

export default AudioListsPanel;
