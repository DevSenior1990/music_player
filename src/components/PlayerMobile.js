import React from 'react'
import cls from 'classnames'
import PlayModeTip from './PlayModeTip'

const prefix = 'react-jinke-music-player-mobile'

const PlayerMobile = ({
  name,
  cover,
  singer,
  playing,
  duration,
  currentTime,
  loading,
  themeSwitch,
  progressBar,
  openAudioListsPanel,
  audioPrevPlay,
  audioNextPlay,
  playMode,
  onClose,
  pause,
  playModeTipVisible,
  currentPlayModeName,
  extendsContent,
  onPlay,
  glassBg,
  LyricIcon,
  onCoverClick,
  autoHiddenCover,
  icon,
}) => (
  <div className={cls(prefix, { 'default-bg': !glassBg, 'glass-bg': glassBg })}>
    <PlayModeTip
      prefix={prefix}
      visible={playModeTipVisible}
      title={playMode}
      text={currentPlayModeName}
    />
    <div className={`${prefix}-header group`}>
      <div className={`${prefix}-header-left`} />
      <div className={`${prefix}-header-title`}>{name}</div>
      <div className={`${prefix}-header-right`} onClick={onClose}>
        {icon.close}
      </div>
    </div>
    <div className={`${prefix}-singer text-center group`}>
      <span className={`${prefix}-singer-name`}>{singer}</span>
    </div>
    <div className={`${prefix}-switch text-center group`}>{themeSwitch}</div>
    {/* lgtm [js/trivial-conditional] */}
    {(!autoHiddenCover || (autoHiddenCover && cover)) && (
      <div
        className={`${prefix}-cover text-center`}
        onClick={() => onCoverClick()}
      >
        <img
          src={cover}
          alt="cover"
          className={cls('cover', {
            'img-rotate-pause': pause || !playing || !cover,
          })}
        />
      </div>
    )}
    <div className={`${prefix}-progress group`}>
      <span className="current-time">{loading ? '--' : currentTime}</span>
      {progressBar}
      <span className="duration text-right">{loading ? '--' : duration}</span>
    </div>
    <div className={`${prefix}-toggle text-center group`}>
      {loading ? (
        icon.loading
      ) : (
        <>
          <span
            className="group prev-audio"
            title="Previous track"
            onClick={audioPrevPlay}
          >
            {icon.prev}
          </span>
          <span
            className="group play-btn"
            title={playing ? 'Click to pause' : 'Click to play'}
            onClick={onPlay}
          >
            {playing ? icon.pause : icon.play}
          </span>
          <span
            className="group next-audio"
            title="Next track"
            onClick={audioNextPlay}
          >
            {icon.next}
          </span>
        </>
      )}
    </div>
    <div className={`${prefix}-operation group`}>
      <ul className="items">
        {[playMode, icon.download, icon.reload, LyricIcon]
          .filter(Boolean)
          .map((item) => (
            <li className="item" key={item.props.className}>
              {item}
            </li>
          ))}
        {extendsContent}
        <li className="item" onClick={openAudioListsPanel}>
          {icon.playLists}
        </li>
      </ul>
    </div>
  </div>
)

PlayerMobile.defaultProps = {
  icon: {},
}

export default PlayerMobile
