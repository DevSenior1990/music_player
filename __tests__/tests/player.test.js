/* global describe,it*/
/*eslint-disable no-console */
//TODOS: 完成点击事件模拟测试 
import React from "react";
import assert from "power-assert";
import { expect } from "chai";
import { shallow, mount } from "enzyme";
// import sinon from 'sinon';

import ReactJkMusicPlayer, { createRandomNum, formatTime } from "../../src";
import PlayerMobile, { PlayModeTip } from "../../src/playerMobile";
import AudioListsPanel from "../../src/audioListsPanel";

describe("<ReactJkMusicPlayer/>", () => {
  it("should render a <ReactJkMusicPlayer/> components", () => {
    const wrapper = mount(<ReactJkMusicPlayer />);
    assert(wrapper.find(".react-jinke-music-player-main").length === 1);
    assert(wrapper.find(".react-jinke-music-player").length >= 1);
  });
  it("should render <AudioListsPanel/> components", () => {
    const wrapper = mount(<ReactJkMusicPlayer />);
    assert(wrapper.find(AudioListsPanel).length === 1);
    assert(wrapper.find(".audio-lists-panel").length === 1);
  });
  it("should find a <PlayerMobile/> components", () => {
    const wrapper = shallow(<PlayerMobile />);
    assert(wrapper.find(".react-jinke-music-player-mobile").length === 1);
    assert(wrapper.find(PlayModeTip).length === 1);
  });
  it("should render mini of full mode", () => {
    const wrapper = mount(<ReactJkMusicPlayer mode="full" />);
    assert(wrapper.props().mode === "full");
    wrapper.setProps({ mode: "mini" });
    assert(wrapper.props().mode === "mini");
  });
  it("should render dark and light theme", () => {
    const wrapper = mount(<ReactJkMusicPlayer theme="dark" />);
    assert(wrapper.props().theme === "dark");
    wrapper.setState({ theme: "dark" });
    assert(wrapper.find(".light-theme").length === 0);
    assert(wrapper.find(".dark-theme").length === 1);
    wrapper.setProps({ theme: "light" });
    wrapper.setState({ theme: "light" });
    assert(wrapper.props().theme === "light");
    assert(wrapper.find(".dark-theme").length === 0);
    assert(wrapper.find(".light-theme").length === 1);
  });
  it('should find audioLists and return Array, playlists show "no music" text ', () => {
    const testProps = {
      audioLists: [
        {
          name: "name",
          singer: "singer",
          cover: "test.jpg",
          musicSrc: "test.mp3"
        }
      ]
    };
    const wrapper = mount(<ReactJkMusicPlayer {...testProps} />);
    assert(wrapper.props().audioLists.length >= 1);
    assert(wrapper.props().audioLists[0].name === "name");
    assert(wrapper.props().audioLists[0].singer === "singer");
    assert(wrapper.props().audioLists[0].cover === "test.jpg");
    assert(wrapper.props().audioLists[0].musicSrc === "test.mp3");
    wrapper.setProps({ audioLists: [] });
    assert(wrapper.props().audioLists.length === 0);
    expect(wrapper.text()).to.contain("no music");
  });
  it("should toggle group setting buttons", () => {
    const wrapper = mount(
      <ReactJkMusicPlayer
        showDownload={true}
        showThemeSwitch={true}
        showReload={true}
        showPlayMode={true}
      />
    );
    wrapper.setState({ loading: false, toggle: true });
    assert(wrapper.find(".audio-download").length >= 1);
    assert(wrapper.find(".group").length >= 1);
    assert(wrapper.find(".theme-switch").length >= 1);
    assert(wrapper.find(".reload-btn").length >= 1);
    assert(wrapper.find(".loop-btn").length === 1);
    wrapper.setState({ toggle: false });
    assert(wrapper.find(".audio-download").length === 0);
    assert(wrapper.find(".theme-switch").length === 0);
    assert(wrapper.find(".reload-btn").length === 0);
    assert(wrapper.find(".loop-btn").length === 0);
  });
  it("should find Specified text", () => {
    const testProps = {
      playModeText: {
        order: "order",
        orderLoop: "orderLoop",
        singleLoop: "singleLoop",
        shufflePlay: "shufflePlay"
      },
      controllerTitle: "controllerTitle",
      openText: "openText",
      closeText: "closeText",
      panelTitle: "panelTitle",
      notContentText: "notContentText",
      checkedText: "checkedText",
      unCheckedText: "unCheckedText",
      showDownload: true,
      showThemeSwitch: true,
      showReload: true,
      showPlayMode: true,
      defaultPlayMode: "order"
    };
    const wrapper = mount(<ReactJkMusicPlayer {...testProps} />);
    expect(wrapper.text()).to.contain("openText");

    wrapper.setState({ toggle: false, loading: false });
    expect(wrapper.text()).to.contain("controllerTitle");

    wrapper.setState({ toggle: true });
    expect(wrapper.text()).to.contain("panelTitle");
    expect(wrapper.text()).to.contain("unCheckedText");

    wrapper.setState({ theme: "light" });
    expect(wrapper.text()).to.contain("checkedText");
    expect(wrapper.text()).to.contain("order");

    wrapper.setProps({ defaultPlayMode: "orderLoop" });
    expect(wrapper.text()).to.contain("orderLoop");

    wrapper.setProps({ defaultPlayMode: "singleLoop" });
    expect(wrapper.text()).to.contain("singleLoop");

    wrapper.setProps({ defaultPlayMode: "shufflePlay" });
    expect(wrapper.text()).to.contain("shufflePlay");

    wrapper.setProps({ audioLists: [] });
    expect(wrapper.text()).to.contain("notContentText");
  });
  it("should render range random", () => {
    const repeat = 10;
    const result = new Array(repeat).fill().map((_, i) => {
      return createRandomNum(0, i + 1) <= i + 1;
    });
    expect(result.filter(v => v)).to.have.length(repeat);
  });
  it("should print second return format time", () => {
    assert(formatTime(30000) === "20:00");
    assert(formatTime(60) === "00:60");
    assert(formatTime(140) === "02:20");
  });
//   it('simulates click events', () => {
//     const onButtonClick = sinon.spy();
//     const wrapper = mount(<ReactJkMusicPlayer onMouseDown={onButtonClick} />);
//     wrapper.find('.react-jinke-music-player').simulate('click');
//     expect(onButtonClick).to.have.property('callCount', 1);
//   });
});
