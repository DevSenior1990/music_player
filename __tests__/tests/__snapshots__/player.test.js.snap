// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactJkMusicPlayer/> should render a <ReactJkMusicPlayer/> components 1`] = `
<div
  class="react-jinke-music-player-main dark-theme text-class-name"
  tabindex="-1"
>
  <div
    class="react-jinke-music-player react-draggable"
    style="left: 0px; top: 0px; transform: translate(0px,0px);"
    tabindex="-1"
  >
    <div
      class="music-player"
    >
      <div
        class="scale music-player-controller"
        id="music-player-controller"
        style="background-image: url();"
      >
        <span
          class="controller-title"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m38.6 19.8q0 3.7-1.3 7l-0.4 1.1-4.2 0.7q-0.5 1.9-2 3.1t-3.5 1.2v0.7q0 0.3-0.2 0.5t-0.5 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5v0.7q1.6 0 2.9 0.8t2.1 2.2l1.5-0.3q0.7-2.1 0.7-4.3 0-3.3-2-6.3t-5.3-4.6-7-1.8-7.1 1.8-5.2 4.6-2 6.3q0 2.2 0.6 4.3l1.6 0.2q0.7-1.3 2-2.1t2.9-0.8v-0.7q0-0.3 0.2-0.5t0.5-0.2h1.5q0.3 0 0.5 0.2t0.2 0.5v12.9q0 0.3-0.2 0.5t-0.5 0.2h-1.5q-0.3 0-0.5-0.2t-0.2-0.5v-0.7q-1.9 0-3.5-1.2t-2-3.1l-4.1-0.7-0.5-1.1q-1.3-3.3-1.3-7 0-3.4 1.5-6.5t4-5.4 5.9-3.7 7.2-1.3 7.1 1.3 6 3.7 4 5.4 1.5 6.5z"
              />
            </g>
          </svg>
        </span>
        <div
          class="music-player-controller-setting"
        >
          Open
        </div>
      </div>
    </div>
  </div>
  <div
    class="audio-lists-panel"
  >
    <div
      class="audio-lists-panel-header"
    >
      <h2
        class="audio-lists-panel-header-title"
      >
        <span>
          Playlists / 
        </span>
        <span
          class="audio-lists-panel-header-num"
        >
          0
        </span>
        <span
          class="audio-lists-panel-header-close-btn"
          title="Close"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
              />
            </g>
          </svg>
        </span>
        <span
          class="audio-lists-panel-header-line"
        />
        <span
          class="audio-lists-panel-header-delete-btn"
          title="Delete audio lists"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m15.9 16.4v12.9q0 0.3-0.2 0.5t-0.5 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m5.7 0v12.9q0 0.3-0.2 0.5t-0.5 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m5.8 0v12.9q0 0.3-0.2 0.5t-0.6 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.4 0 0.6 0.2t0.2 0.5z m2.8 16.2v-21.2h-20v21.2q0 0.5 0.2 0.9t0.3 0.6 0.2 0.2h18.6q0.1 0 0.2-0.2t0.4-0.6 0.1-0.9z m-15-24h10l-1.1-2.6q-0.1-0.2-0.3-0.3h-7.1q-0.2 0.1-0.4 0.3z m20.7 0.7v1.4q0 0.3-0.2 0.5t-0.5 0.2h-2.1v21.2q0 1.8-1.1 3.2t-2.5 1.3h-18.6q-1.4 0-2.5-1.3t-1-3.1v-21.3h-2.2q-0.3 0-0.5-0.2t-0.2-0.5v-1.4q0-0.3 0.2-0.5t0.5-0.2h6.9l1.6-3.8q0.3-0.8 1.2-1.4t1.7-0.5h7.2q0.9 0 1.8 0.5t1.2 1.4l1.5 3.8h6.9q0.3 0 0.5 0.2t0.2 0.5z"
              />
            </g>
          </svg>
        </span>
      </h2>
    </div>
    <div
      class="audio-lists-panel-content no-content"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style="vertical-align: middle;"
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m6.6 10v23.4h23.4v3.2h-23.4c-1.8 0-3.2-1.4-3.2-3.2v-23.4h3.2z m23.4 1.6v-3.2h-6.6v9.1c-0.7-0.6-1.6-0.9-2.5-0.9-2.3 0-4.3 2-4.3 4.3s2 4.1 4.3 4.1 4.1-1.9 4.1-4.1v-9.3h5z m3.4-8.2c1.8 0 3.2 1.4 3.2 3.2v20c0 1.8-1.4 3.4-3.2 3.4h-20c-1.8 0-3.4-1.6-3.4-3.4v-20c0-1.8 1.6-3.2 3.4-3.2h20z"
            />
          </g>
        </svg>
      </span>
      <span
        class="no-data"
      >
        No music
      </span>
    </div>
  </div>
  <div
    class="play-mode-title"
  >
    Play in order
  </div>
  <audio
    class="music-player-audio"
    src=""
    title=" "
  />
</div>
`;
